### ISSUE TITLE
Failed to find a valid digest in the 'integrity' attribute for javascript files

### ISSUE THREAD

**Issue by glxgxr** [lhapaipai/vite-bundle#80]
### vite-bundle version

6.3.4

### vite-plugin-symfony version

6.3.4

### your OS, Symfony version, PHP version, Node version

OS Linux, Docker, Symfony 5.4.28, Node v21.6.1

### Description

Using hashing algorithm for integrity check currently fails for javacript files. It does not matter whether it is a CDN or local files.
When I check the integrity hashes on my public CDN with tools like https://www.srihash.org/, it calculates the same integriy digest, as the browser console compains about:

> Failed to find a valid digest in the 'integrity' attribute for resource 'https://app.blabla.local/build/assets/appScripts-EqTxXgdP.js' with computed SHA-256 integrity 'ljVss2YLNlzXj3vEVZ....'. The resource has been blocked.

All hashes for the css files seem to work fine, except the one for javascript. The hash just doesn't match with the one in the `entrypoints.json`.

Probably related to https://github.com/lhapaipai/vite-bundle/issues/65

### How to reproduce

This is the config:

```js
import {defineConfig, loadEnv} from 'vite';
import symfonyPlugin from 'vite-plugin-symfony';
import vue from '@vitejs/plugin-vue';

//eslint-disable-next-line no-unused-vars
export default defineConfig(({command, mode}) => {
    //eslint-disable-next-line no-undef
    const env = loadEnv(mode, process.cwd(), '');

    return {
        base: 'https://my-cdn/build/',
        plugins: [
            nodeResolve(),
            vue(),
            viteStaticCopy({
                targets: [
                    {
                        src: './assets/images',
                        dest: './',
                    },
                    {
                        src: './assets/videos',
                        dest: './',
                    },
                ],
            }),
            symfonyPlugin({
                servePublic: 'public',
                refresh: true,
                sriAlgorithm: "sha256"
            })
        ],
        build: {
            root: './assets',
            manifest: true,
            rollupOptions: {
                input: {
                    app: './assets/js/app.js',
                    leaflet: './assets/js/leaflet.js',
                },
                output: {
                    manualChunks: {
                        vue: ['vue']
                    }
                }
            },
            define: {
                __APP_ENV__: env.APP_ENV || 'dev',
            },
            outDir: 'public/build',
            modulePreload: {
                polyfill: true,
            },
        },
    };
});

```

app.html.twig
```twig
    {{ vite_entry_script_tags('app') }}
```

__Update__:

After hours of debugging, I've found out, that similar to the issue mentioned above, the files BEFORE the hash is getting calculated, has `__VITE_PRELOAD__` placeholders. The finale compiled files have `vite.mapDeps(...)` instead of the placeholders. That why I think the hashes are wrong.

**Comment by darrylhein** [comment]
I've been having a similar issue, but hadn't had the time to debug.

**Comment by glxgxr** [comment]
Okay, aftter one more hours of debugging, I've found out that the problem is most likely cause by dynamic imports. I've removed all dynamic imports (which is definetely not a soluation), so that there are no more placeholders and `vite.mapDeps()` functions in the compiled code.
Maybe this can help to investiage this issue further.

**Comment by lhapaipai** [comment]
indeed Vite lets us define plugins upstream and downstream, but it is still he who chooses to place his plugins at the end...
https://github.com/vitejs/vite/blob/76f30ae23b92f9af910ec02d98e2baaefa12141f/packages/vite/src/node/plugins/index.ts#L99

```ts
// vite/src/node/build.ts
export async function resolveBuildPlugins(config: ResolvedConfig): Promise<{
  pre: Plugin[]
  post: Plugin[]
}> {
  // ...
  return {
    // ...
    post: [
      buildImportAnalysisPlugin(config),
      ...(config.esbuild !== false ? [buildEsbuildPlugin(config)] : []),
      ...(options.minify ? [terserPlugin(config)] : []),
      ...(!config.isWorker
        ? [
            ...(options.manifest ? [manifestPlugin(config)] : []),
            ...(options.ssrManifest ? [ssrManifestPlugin(config)] : []),
            buildReporterPlugin(config),
          ]
        : []),
      loadFallbackPlugin(),
    ],
  }
}
```

I had to force the plugin to run almost at the very last. this should be fixed with v6.4.0
